python/ray/_private/runtime_env/nsight.py

```diff
@@ -71,7 +71,8 @@ async def _check_nsight_script(
         nsight_config_copy["o"] = str(Path(self._nsight_dir) / "empty")
         nsight_cmd = parse_nsight_config(nsight_config_copy)
         try:
-            nsight_cmd = nsight_cmd + ["python", "-c", '""']
+            python_executable = sys.executable or "python"
```

Comment:
We can assume that `sys.executable` is available, since we do so in many other places

I believe it would only not be available if the script is executed with a statically or dynamically linked python library, which is a very exotic setting (and even if a user uses this, they could easily set sys.executable to a proper interpreter before they invoke the python script).

Reply:
Ah makes sense -- I don't think that's a problem, the official type stub for sys.executable seems to be str: https://github.com/python/typeshed/blob/567b488fc28978642bfeb70b29f11453b0ff124c/stdlib/sys/__init__.pyi#L31